src/riak_kv_pb_timeseries.erl

```diff
@@ -255,6 +261,108 @@ wait_until_active(Table, State, Seconds) ->
 %% functions called from check_table_and_call, one per ts* request
 %% ---------------------------------------------------
 
+
+%%
+%% INSERT statements, called from check_table_and_call.
+%%
+-spec make_insert_response(module(), #riak_sql_insert_v1{}) ->
+                           #tsqueryresp{} | #rpberrorresp{}.
+make_insert_response(Mod, #riak_sql_insert_v1{'INSERT' = Table, fields = Fields, values = Values}) ->
+    case lookup_field_positions(Mod, Fields) of
+	{error, FieldReason} ->
+	    make_rpberrresp(?E_BAD_QUERY, FieldReason);
+	{ok, Positions} ->
+        Empty = make_empty_row(Mod),
+        case xlate_insert_to_putdata(Values, Positions, Empty) of
+            {error, ValueReason} ->
+                make_rpberrresp(?E_BAD_QUERY, ValueReason);
+            {ok, Data} ->
+                Response = sub_putreq_common(Mod, Table, Data, #state{}),
```

Comment:
You are producing a default State for `sub_putreq_common`, which happens to be OK because there's no useful fields in `#state{}` for the put request, only to be discarded in `xlate_put_to_qry_resp`. I would suggest simply copying the outer case statement from `sub_putreq_common` right here, or perhaps as a separate local helper function, which would produce proper `#tsqueryresp{}` instead of `#tsputresp{}`.